### PR TITLE
[ADL] Update TPM event logging to match BIOS

### PR DESCRIPTION
**Pull request by sean-m-mcginn**
If measured boot disabled via BtG profile but enabled via SBL
config flag, skip logging startup locality TPM event

If measured boot enabled via BtG profile or SBL config flag, log
CRTM version TPM event

### CODE REVIEW

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c [hunk 1]
```diff
@@ -376,18 +377,30 @@ PlatformFeaturesInit (
     HeciBaseAddress = MeGetHeciMmPciAddress (0, 0);
     GetBootGuardInfo (HeciBaseAddress, &PlatformData->BtGuardInfo);
     DEBUG ((DEBUG_INFO, "GetPlatformDataPtr is copied 0x%08X \n", PlatformData));
-    if ((PlatformData->BtGuardInfo.MeasuredBoot) && (!PlatformData->BtGuardInfo.BypassTpmInit)) {
-      // Disable the measured boot when TPM init is not succesful by ACM
-      LdrFeatures &= ~FEATURE_MEASURED_BOOT;
```
Comment (lsubash) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c:381]: Check when TPM init is not successful by BTG.

File: BootloaderCommonPkg/Library/TpmLib/TpmLib.inf [hunk 2]
```diff
@@ -37,6 +37,7 @@
 [Packages]
   MdePkg/MdePkg.dec
   BootloaderCommonPkg/BootloaderCommonPkg.dec
+  BootloaderCorePkg/BootloaderCorePkg.dec
```
Comment (gdong1) [inline BootloaderCommonPkg/Library/TpmLib/TpmLib.inf:40]: BootloaderCommonPkg should not depend on BootloaderCorePkg (CorePkg could depend on CommonPkg).

File: BootloaderCommonPkg/Include/Library/TpmLib.h [hunk 3]
```diff
@@ -16,6 +16,17 @@
...
 #define ACPI_SSDT_TPM2_DEVICE_OEM_TABLE_ID  SIGNATURE_64('T', 'p', 'm', '2', 'T', 'a', 'b', 'l')
 
+/**
+  Extend SBL version in PCR [0].
+
+  @retval RETURN_SUCCESS      Operation completed successfully.
+  @retval Others              Unable to extend PCR.
+**/
+RETURN_STATUS
+TpmLogCrtmVersionEvent(
```
Comment (gdong1) [inline BootloaderCommonPkg/Include/Library/TpmLib.h:26]: It looks all the library API functions in this file, EFIAPI is missing. Please add it for both header file and implementation.

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c [hunk 4]
```diff
@@ -441,26 +441,32 @@ typedef struct {
...
+    UINT64 MemScrubPolicy     : 2;      // 25-26 Memory scrubbing policy
+    UINT64 Reserved2          : 2;      // 27-28
+    UINT64 DmaProtection      : 1;      // 29    DMA Protection
+    UINT64 Reserved3          : 2;      // 30-31
+    UINT64 SCrtmStatus        : 3;      // 32-34 S-CRTM status
+    UINT64 Cosign             : 1;      // 35    CPU co-signing
+    UINT64 TpmStartupLocality : 1;      // 36    TPM startup locality.
+    UINT64 Reserved           :27;      // 37-63
   } Bits;
-  UINT32 Data;
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c:463]: why we are updating this structure?

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c [hunk 5]
```diff
@@ -1268,14 +1301,41 @@ CreateTpmEventLog (
...
+    // create event logs from previous PCR extensions on behalf of ACM.
+    //
+    if (AcmPolicySts.Bits.SCrtmStatus != 0) {
+      CreateDetailPcrEvent (ActivePcrBanks);
+    }
 
-    CreateLocalityStartupEvent (TpmType, ActivePcrBanks);
-    CreateDetailPcrEvent (TpmType, ActivePcrBanks);
-    CreateAuthorityPcrEvent (TpmType, ActivePcrBanks);
+    CreateAuthorityPcrEvent (ActivePcrBanks);
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c:1338]: Authority PCR also should have SCRTSTatus check.

File: BootloaderCommonPkg/Library/TpmLib/TpmLib.inf [hunk 6]
```diff
@@ -54,6 +54,7 @@
   ResetSystemLib
   HobLib
   HobBuildLib
+  BootloaderCoreLib
```
Comment (gdong1) [inline BootloaderCommonPkg/Library/TpmLib/TpmLib.inf:57]: I think BootloaderCoreLib is not required to add.

**Review by gdong1 — APPROVED** [review]

**Review by lsubash — COMMENTED** [review]
Sean,would you split the changes in two patches?
Changes to eventlog lib are common to all platforms. 
For TGL/ADL/RPL, it looks fine. AcmPolicySts Struct aligns.
For EHL, CML platforms we need to check how this AcmPolicySts gets populated.

**Review by lsubash — APPROVED** [review]
looks good

File: BootloaderCommonPkg/Include/Library/TpmLib.h
Comment (kvparikh89) [inline BootloaderCommonPkg/Include/Library/TpmLib.h:1]: Update  header year

File: BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c
Comment (kvparikh89) [inline BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c:1]: Update header year

File: BootloaderCommonPkg/Library/TpmLib/TpmLib.c
Comment (kvparikh89) [inline BootloaderCommonPkg/Library/TpmLib/TpmLib.c:1]: Update  header year

File: BootloaderCommonPkg/Library/TpmLib/TpmLib.inf
Comment (kvparikh89) [inline BootloaderCommonPkg/Library/TpmLib/TpmLib.inf:1]: Update  header year

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c [hunk 7]
```diff
@@ -368,6 +367,8 @@ PlatformFeaturesInit (
     if (FeaturesCfgData->Features.MeasuredBoot == 0) {
       LdrFeatures &= ~FEATURE_MEASURED_BOOT;
     }
+  } else {
```
Comment (kvparikh89) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c:370]: Set pointer to NULL

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c [hunk 8]
```diff
@@ -368,6 +367,8 @@ PlatformFeaturesInit (
...
+    DEBUG((DEBUG_INFO, "FEATURES_CFG_DATA is NULL"));
```
Comment (kvparikh89) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c:371]: Refer other debug prints and edit for consistency

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c [hunk 9]
```diff
@@ -399,48 +412,50 @@ TpmInitialize (
   EFI_STATUS                   Status;
   UINT8                        BootMode;
   PLATFORM_DATA               *PlatformData;
```
Comment (kvparikh89) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c:414]: Set pointer to NULL

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c [hunk 10]
```diff
@@ -399,48 +412,50 @@ TpmInitialize (
...
-        ResetSystem(EfiResetCold);
-        CpuDeadLoop ();
-      }
-
-      // TPM is initialized in by ACM.
-      Status = TpmInit(PlatformData->BtGuardInfo.BypassTpmInit, BootMode);
-    } else {
-      // Initialize TPM if it has not already been initialized by BootGuard component (i.e. ACM)
-      Status = TpmInit(FALSE, BootMode);
+  if (PlatformData->BtGuardInfo.BootGuardCapability &&
```
Comment (kvparikh89) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c:430]: Is NULL check required?

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c
Comment (kvparikh89) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c:1]: Update header year

**Review by kvparikh89 — CHANGES_REQUESTED** [review]

**Pull request merged**